components/tools/OmeroWeb/test/unit/test_util.py

```diff
@@ -53,3 +58,34 @@ def test_get_date_time(self):
             getDateTime("invalid")
         with pytest.raises(ValueError):
             getDateTime("2015-12-01")
+
+    @pytest.mark.parametrize('top_links', [(
+        ('{"viewname": "load_template", "args": ["userdata"],'
+         '"query_string": {"experimenter": -1}}'),
+        "/webclient/userdata/?experimenter=-1"),
+        ('{"viewname": "webindex", "query_string": {"foo": "bar"}}',
+         "/webclient/?foo=bar"),
+        ('{"viewname": "foo", "args": ["bar"]}', ""),
+        ('{"viewname": "foo", "query_string": {"foo": "bar"}}', ""),
+        ])
+    def test_reverse_with_params_dict(self, top_links):
+        top_link = json.loads(top_links[0])
+        assert reverse_with_params(**top_link) == top_links[1]
+
+    @pytest.mark.parametrize('top_links', [
+        ("history", "/webclient/history/"),
+        ("webindex", "/webclient/"),
+        ])
+    def test_reverse_with_params_string(self, top_links):
+        top_link = top_links[0]
+        assert reverse_with_params(top_link) == reverse(top_link) \
+            == top_links[1]
+
+    @pytest.mark.parametrize('top_link', ["foo", '', None])
+    def test_bad_reverse_with_params_string(self, top_link):
+        try:
```

Comment:
I would do this with pytest's expected exception handling:

 * http://pytest.org/latest/assert.html#assertions-about-expected-exceptions